src/components/multipart/downloadButton.tsx

```diff
@@ -0,0 +1,58 @@
+import IconButton from '@mui/material/IconButton'
+import Tooltip from '@mui/material/Tooltip'
+import React from 'react'
+
+import Icon from '../icon/icon'
+import type { GetAuthHeaders } from '../types'
+
+type DownloadButtonProps = {
+  url: string
+  fileName: string
+  getAuthHeaders?: GetAuthHeaders
+}
+
+export default function DownloadButton(props: DownloadButtonProps) {
+  function handleAuthenticatedDownload() {
+    props.getAuthHeaders!()
+      .then((headersObj: Record<string, any>) => {
+        const headers = headersObj.headers
+        return fetch(props.url, { headers })
+      })
+      .then((response) => {
+        return response.blob()
+      })
```

Comment:
This can be simplified using axios

```
  // Use axios to fetch the file
      return axios.get(props.url, {
        headers,
        responseType: 'blob', // Important to specify 'blob' to handle file downloads
      });
    })
    .then((response) => {
      const blob = response.data

      // Create a download link and trigger the download
      const downloadUrl = URL.createObjectURL(blob)